### PR TITLE
Work in Progress: deci yolo rf100

### PR DESCRIPTION
**Pull request by BloodAxe**
✅ Pretrain weights
✅ Integration tests
✅ RF-100 recipe 
✅ Docs
✅ License

### CODE REVIEW

**Review by ofrimasad — DISMISSED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -176,10 +184,10 @@ model = models.get("model-name", pretrained_weights="pretrained-model-name")
 #### Transfer Learning 
   <table class="tfo-notebook-buttons" align="left">
  <td width="500">  
-  <a target="_blank" href="https://bit.ly/3xzIutb"><img src="./docs/assets/SG_img/colab_logo.png" /> Classification Transfer Learning</a>
+  <a target="_blank" href="https://bit.ly/3xzIutb"><img style="vertical-align:middle" src="./docs/assets/SG_img/colab_logo.png" /> Classification Transfer Learning</a>
```
Comment (shaydeci) [inline README.md:187]: Are all these readme.md changes on purpose?

File: scripts/Dockerfile.branch [hunk 2]
```diff
@@ -0,0 +1,15 @@
+ARG DOCKER_IMAGE_TAG=11.3.1-devel-ubuntu20.04
```
Comment (shaydeci) [inline scripts/Dockerfile.branch:1]: Im pretty confused...there seem to be some files and changes from infra team that are not in master but are in our YoloNas branch...

File: scripts/Dockerfile.branch.code [hunk 3]
```diff
@@ -0,0 +1,10 @@
+ARG BASE_TAG
```
Comment (shaydeci) [inline scripts/Dockerfile.branch.code:1]: Same as previous comment.

File: src/super_gradients/recipes/coco2017_yolo_nas_s.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+#   1. Move to the project root (where you will find the ReadMe and src folder)
+#   2. Run the command you want:
+#         yolo_nas_s: python src/super_gradients/examples/train_from_recipe_example/train_from_recipe.py --config-name=coco2017_yolo_nas_s
+#
+
+defaults:
+  - training_hyperparams: coco2017_yolo_nas_train_params
+  - dataset_params: coco_detection_yolo_nas_dataset_params
+  - arch_params: yolo_nas_s_arch_params
+  - checkpoint_params: default_checkpoint_params
```
Comment (shaydeci) [inline src/super_gradients/recipes/coco2017_yolo_nas_s.yaml:16]: Can you update with the current variable_setup inheritance for shortcuts ?

File: src/super_gradients/recipes/coco2017_yolo_nas_s.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+architecture: yolo_nas_s
+
+multi_gpu: DDP
+num_gpus: 8
+
+experiment_suffix: ""
+experiment_name: coco2017_${architecture}${experiment_suffix}
+
+ckpt_root_dir:
```
Comment (shaydeci) [inline src/super_gradients/recipes/coco2017_yolo_nas_s.yaml:44]: Following my previous comment, we can get rid of ckpt_root_dir and the hydra:.. part.

File: src/super_gradients/recipes/dataset_params/coco_detection_yolo_nas_dataset_params.yaml [hunk 6]
```diff
@@ -67,13 +67,13 @@ val_dataset_params:
...
         pad_value: 114
-    - DetectionStandardizeImage:
+    - DetectionStandardize:
         max_value: 255.
     - DetectionImagePermute
     - DetectionTargetsFormatTransform:
         max_targets: 50
         input_dim: [640, 640]
-        output_format: LABEL_NORMALIZED_CXCYWH
+        output_format: LABEL_CXCYWH
```
Comment (shaydeci) [inline src/super_gradients/recipes/dataset_params/coco_detection_yolo_nas_dataset_params.yaml:76]: Nice catch!

File: src/super_gradients/recipes/roboflow_yolo_nas_m.yaml [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+# `dataset_name` refers to the official name of the dataset.
+# You can find it in the url of the dataset: https://universe.roboflow.com/roboflow-100/digits-t2eg6 -> digits-t2eg6
+#
+# Example: python -m super_gradients.train_from_recipe --config-name=roboflow_yolo_nas_m dataset_name=digits-t2eg6
+
+defaults:
+  - training_hyperparams: coco2017_yolo_nas_train_params
+  - dataset_params: roboflow_detection_dataset_params
+  - checkpoint_params: default_checkpoint_params
+  - arch_params: yolo_nas_m_arch_params
```
Comment (shaydeci) [inline src/super_gradients/recipes/roboflow_yolo_nas_m.yaml:13]: Same as my comment in src/super_gradients/recipes/coco2017_yolo_nas_s.yaml defaults.

File: src/super_gradients/recipes/roboflow_yolo_nas_m.yaml [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+multi_gpu: Off
+num_gpus: 1
+experiment_suffix: ""
+experiment_name: ${architecture}_roboflow_${dataset_name}${experiment_suffix}
+ckpt_root_dir:
+
+
+# THE FOLLOWING PARAMS ARE DIRECTLY USED BY HYDRA
+hydra:
```
Comment (shaydeci) [inline src/super_gradients/recipes/roboflow_yolo_nas_m.yaml:95]: Same as my comment in src/super_gradients/recipes/coco2017_yolo_nas_s.yaml defaults.

File: src/super_gradients/recipes/roboflow_yolo_nas_s.yaml [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+# `dataset_name` refers to the official name of the dataset.
+# You can find it in the url of the dataset: https://universe.roboflow.com/roboflow-100/digits-t2eg6 -> digits-t2eg6
+#
+# Example: python -m super_gradients.train_from_recipe --config-name=roboflow_yolo_nas_s dataset_name=digits-t2eg6
+
+defaults:
+  - training_hyperparams: coco2017_yolo_nas_train_params
+  - dataset_params: roboflow_detection_dataset_params
+  - checkpoint_params: default_checkpoint_params
+  - arch_params: yolo_nas_s_arch_params
```
Comment (shaydeci) [inline src/super_gradients/recipes/roboflow_yolo_nas_s.yaml:13]: Same as my comment in src/super_gradients/recipes/coco2017_yolo_nas_s.yaml defaults.

File: src/super_gradients/recipes/roboflow_yolo_nas_s.yaml [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+
+multi_gpu: Off
+num_gpus: 1
+experiment_suffix: ""
+experiment_name: ${architecture}_roboflow_${dataset_name}${experiment_suffix}
+ckpt_root_dir:
+
+
+# THE FOLLOWING PARAMS ARE DIRECTLY USED BY HYDRA
+hydra:
```
Comment (shaydeci) [inline src/super_gradients/recipes/roboflow_yolo_nas_s.yaml:95]: Same as my comment in src/super_gradients/recipes/coco2017_yolo_nas_s.yaml defaults.

File: src/super_gradients/training/datasets/detection_datasets/roboflow/metadata.py [hunk 11]
```diff
@@ -33,7 +33,7 @@
     "underwater-objects-5v7p8": {"category": "underwater", "train": 5320, "test": 760, "valid": 1520, "size": 7600, "num_classes": 5, "num_classes_found": 5},
     "coral-lwptl": {"category": "underwater", "train": 427, "test": 74, "valid": 93, "size": 594, "num_classes": 14, "num_classes_found": 14},
     "tweeter-posts": {"category": "documents", "train": 87, "test": 9, "valid": 21, "size": 117, "num_classes": 2, "num_classes_found": 2},
-    "tweeter-profile": {"category": "documents", "train": 425, "test": 61, "valid": 121, "size": 607, "num_classes": 1, "num_classes_found": 0},
+    "tweeter-profile": {"category": "documents", "train": 425, "test": 61, "valid": 121, "size": 607, "num_classes": 1, "num_classes_found": 1},
```
Comment (shaydeci) [inline src/super_gradients/training/datasets/detection_datasets/roboflow/metadata.py:36]: I thought this should always stay the same as it is generated once...@Louis-Dupont is this because of the extra class thing we spoke of? in that case should it be changed here ?

**Review by shaydeci — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (BloodAxe) [reply README.md:187]: It hells in PyCharm preview to align the icons with respect to the text, but GitHub strips this out anyway so I will remove it

File: src/super_gradients/modules/interfaces.py [hunk 12]
```diff
@@ -0,0 +1,31 @@
+import abc
+from typing import Callable
+
+from torch import nn
+
+__all__ = ["SupportsReplaceNumClasses"]
+
+
+class SupportsReplaceNumClasses:
```
[the remark this answers is not in the document]
Comment (BloodAxe) [reply src/super_gradients/modules/interfaces.py]: Thanks. Moved

File: src/super_gradients/recipes/coco2017_yolo_nas_s.yaml (hunk 5 above)
Comment (BloodAxe) [reply src/super_gradients/recipes/coco2017_yolo_nas_s.yaml]: Updated in https://github.com/Deci-AI/super-gradients/commit/0d3f52b292883d11c25b1f81ccca427742688653

File: src/super_gradients/recipes/roboflow_yolo_nas_m.yaml (hunk 7 above)
Comment (BloodAxe) [reply src/super_gradients/recipes/roboflow_yolo_nas_m.yaml:13]: Updated in https://github.com/Deci-AI/super-gradients/commit/0d3f52b292883d11c25b1f81ccca427742688653

File: src/super_gradients/recipes/roboflow_yolo_nas_m.yaml (hunk 8 above)
Comment (BloodAxe) [reply src/super_gradients/recipes/roboflow_yolo_nas_m.yaml]: Updated in https://github.com/Deci-AI/super-gradients/commit/0d3f52b292883d11c25b1f81ccca427742688653

File: src/super_gradients/recipes/roboflow_yolo_nas_s.yaml (hunk 9 above)
Comment (BloodAxe) [reply src/super_gradients/recipes/roboflow_yolo_nas_s.yaml:13]: Updated in https://github.com/Deci-AI/super-gradients/commit/0d3f52b292883d11c25b1f81ccca427742688653

File: src/super_gradients/recipes/roboflow_yolo_nas_s.yaml (hunk 10 above)
Comment (BloodAxe) [reply src/super_gradients/recipes/roboflow_yolo_nas_s.yaml]: Updated in https://github.com/Deci-AI/super-gradients/commit/0d3f52b292883d11c25b1f81ccca427742688653

File: .circleci/config.yml [hunk 13]
```diff
@@ -1,6 +1,12 @@
 version: 2.1
 
 parameters:
+  ad_hoc_container_build:
```
[the remark this answers is not in the document]
Comment (BloodAxe) [reply .circleci/config.yml:4]: There are additional CI targets to build & deploy container from CircleCI. This enables us to launch training in cloud using argo.

File: scripts/Dockerfile.branch (hunk 2 above)
Comment (BloodAxe) [reply scripts/Dockerfile.branch:1]: Should I move these changes to another PR or is it ok to leave it here?

Comment (shaydeci) [reply scripts/Dockerfile.branch:1]: Would appreciate if you do, but dont have to.

**Comment by BloodAxe** [comment]
Closing as this PR contains unsigned commits.
New pr https://github.com/Deci-AI/super-gradients/pull/875
